spec/requests/npq_separation/admin/applications/revert_to_pending_controller_spec.rb

```diff
@@ -0,0 +1,66 @@
+require "rails_helper"
+
+RSpec.describe NpqSeparation::Admin::Applications::RevertToPendingController, :ecf_api_disabled, type: :request do
+  include Helpers::NPQSeparationAdminLogin
+
+  subject { response }
+
+  let(:application) { create(:application, :accepted) }
+
+  context "when logged in" do
+    before { sign_in_as_admin }
+
+    describe "#new" do
+      before { get new_npq_separation_admin_applications_revert_to_pending_path(application) }
+
+      it { is_expected.to have_http_status :success }
+      it { is_expected.to have_attributes body: /change the status to pending/i }
+    end
+
+    describe "#create" do
+      before do
+        post npq_separation_admin_applications_revert_to_pending_path(application, params:)
+      end
+
+      context "with valid form params" do
+        let :params do
+          { applications_revert_to_pending: { change_status_to_pending: "yes" } }
+        end
+
+        it { is_expected.to redirect_to npq_separation_admin_application_path(application) }
+      end
+
+      context "with invalid form params" do
+        let :params do
+          { statements_payment_authorisation: { change_status_to_pending: "no" } }
+        end
+
+        it { is_expected.to have_http_status :unprocessable_entity }
+        it { is_expected.to have_attributes body: /change the status to pending/i }
```

Comment:
shouldn't this redirect back to the original page 🤔

Reply:
No - it needs to show the form again with the error message(s)? This is the invalid scenario?

Reply:
Ok, changed to allow the user to select no